File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.java
import java.util.*;

class Solution {
    public int solution(String s) {
        int answer = 0;
        Stack<Character> stack = new Stack<>();
        
        int len = s.length();
        for(int i = 0; i < len; i++){
            char temp = s.charAt(i);
            if(stack.size() == 0){
                stack.add(temp);
            } else if(stack.peek() == temp){
                stack.add(temp);
            } else if(stack.peek() != temp){
                stack.pop();
            }
            if(stack.size() == 0){
                answer++;
            }
        }
        if(stack.size() > 0) answer++;
        return answer;
    }
}

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.js
function solution(s) {
    let answer = 0;
    let stack = [];
    for(x of s){
        stack.push(x);
        let std = stack.filter(e => e == stack[0])
        let dif = stack.filter(e => e != stack[0])
        
        if(std.length * 2== stack.length){
            answer++;
            stack = []
        }
    }
    if(stack.length > 0) answer++;
    return answer;
}